src/getPage.tsx

```diff
@@ -53,83 +56,85 @@ export default async function getPage({
     router,
     useApp,
     useDocument,
+    nonIsolatedModules,
   };
+
   validateOptions(optionsWithDefaults);
   loadNextConfig({ nextRoot });
-  setNextRuntimeConfig({ runtimeEnv: 'client' });
+  setNextRuntimeConfig({ runtimeEnv: RuntimeEnvironment.CLIENT });
 
   const options: ExtendedOptions = {
     ...optionsWithDefaults,
     pagesDirectory: findPagesDirectory({ nextRoot }),
     pageExtensions: getPageExtensions(),
-    env: 'server',
+    env: RuntimeEnvironment.SERVER,
   };
   // @TODO: Consider printing extended options value behind a debug flag
 
   const headManager = useDocument && initHeadManager();
 
   const makePage = async (
-    optionsOverride?: Partial<ExtendedOptions>
-  ): Promise<Page> => {
-    const mergedOptions = { ...options, ...optionsOverride };
-    let { pageElement, pageData, pageObject } = await makePageElement({
-      options: mergedOptions,
+    options: ExtendedOptions
+  ): Promise<MakePageResult> => {
+    const { pageData, pageObject } = await getPageInfo({ options });
+    const { AppComponent, PageComponent, routeData } = makePageConstructs({
+      pageObject,
+      env: options.env,
     });
 
-    if (useDocument && mergedOptions.env === 'client' && headManager) {
-      pageElement = (
+    const pageElement = (
+      <AppComponent Component={PageComponent} pageProps={pageData.props} />
+    );
+
+    if (
+      !useDocument ||
+      options.env !== RuntimeEnvironment.CLIENT ||
+      !headManager
+    ) {
```

Comment:
I'm getting sleepy :) Why are we inverting conditional logic here?

Reply:
I think I changed that back. Probably shortly after you started you review.